### PR TITLE
Add ebuild for NeXus

### PR DESCRIPTION
**Pull request by chatcannon**
A library for the HDF5-based NeXus format for neutron and X-ray scattering data. Depends on minixml, which has been added in a separate PR

### CODE REVIEW

File: dev-libs/minixml/minixml-2.9.ebuild [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+inherit eutils
```
Comment (jlec) [inline dev-libs/minixml/minixml-2.9.ebuild]: Not used, so it can be dropped.

File: dev-libs/minixml/minixml-2.9.ebuild [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+SRC_URI="http://www.msweet.org/files/project3/mxml-${PV}.tar.gz"
+
+LICENSE="LGPL-2-with-linking-exception"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+
+S="${WORKDIR}/mxml-${PV}"
+
+src_install() {
```
Comment (jlec) [inline dev-libs/minixml/minixml-2.9.ebuild]: drop this, as it is the default implementation.

File: sci-libs/nexus/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+<herd>sci</herd>
+<maintainer>
+	<email>cjk34@cam.ac.uk</email>
+	<name>Chris Kerr</name>
+	<description>I don't develop this software, I'm just a Gentoo user who wanted to install it</description>
+</maintainer>
+<longdescription>Libraries and tools for the NeXus data format, developed collaboratively by scientists from major scientific facilities to facilitate analysis and visualisation of neutron, x-ray and muon data. NeXus is based on HDF5.</longdescription>
```
Comment (jlec) [inline sci-libs/nexus/metadata.xml]: wrap long line at 80 chars.

File: sci-libs/nexus/metadata.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+<use>
+	<flag name="xml">Allow using the verbose XML representation of NeXus data</flag>
+	<flag name="doc">Generate library documentation with <pkg>app-doc/doxygen</pkg></flag>
+	<flag name="swig">Generate bindings with <pkg>dev-lang/swig</pkg></flag>
+	<flag name="cbflib">No idea what this does, sorry - it's listed in ./configure --help</flag>
```
Comment (jlec) [inline sci-libs/nexus/metadata.xml]: probably support for cbf images -> sci-lisb/cbflib

File: sci-libs/nexus/nexus-4.3.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+FORTRAN_NEEDED=fortran
+FORTRAN_STANDARD=90
+PYTHON_COMPAT=( python{2_5,2_6,2_7} )
```
Comment (jlec) [inline sci-libs/nexus/nexus-4.3.1.ebuild]: drop python 2.5 and 2.6, both are deprecated and unsupported.

File: dev-libs/minixml/minixml-2.9.ebuild (hunk 2 above)
Comment (chatcannon) [inline dev-libs/minixml/minixml-2.9.ebuild]: I removed the src_install and the install phase no longer works

File: sci-libs/nexus/nexus-4.3.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+		$(use_with xml) \
+		$(use_with cbflib) \
+		$(use_with guile) \
+		$(use_with java) \
+		$(use_with python)
+}
+
+src_compile() {
+	# Handling of dependencies between Fortran module files doesn't play well with parallel make
+	use fortran && MAKEOPTS+=" -j1 "
```
Comment (junghans) [inline sci-libs/nexus/nexus-4.3.1.ebuild]: Move this line into `pkg_setup` and drop `src_compile`.
Also you should explicitly `export  MAKEOPTS=${MAKEOPTS} -j1` as there are profile in which `MAKEOPTS` isn't set (see https://bugs.gentoo.org/show_bug.cgi?id=492372).

**Comment by junghans** [comment]
I am not sure why `mini-xml` is part of this review....

**Comment by chatcannon** [comment]
because NeXus depends on minixml

**Comment by chatcannon** [comment]
I have submitted a new PR #350 for Mini-XML / mxml. I will resubmit this one once that PR is accepted.

**Pull request merged**